Shared/Samples/Query map image sublayer/QueryMapImageSublayerView.swift

```diff
@@ -0,0 +1,183 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct QueryMapImageSublayerView: View {
+    /// The view model for this sample.
+    @State private var model = Model()
+    
+    /// The current viewpoint of the map view.
+    @State private var viewpoint: Viewpoint? = .westernUSA
+    
+    /// A Boolean value indicating whether there is an ongoing query operation.
+    @State private var isQuerying = false
+    
+    /// The minimum population value in the text field.
+    @State private var minimumPopulation: Int?
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    var body: some View {
+        MapView(map: model.map, viewpoint: viewpoint, graphicsOverlays: [model.graphicsOverlay])
+            .onViewpointChanged(kind: .boundingGeometry) { viewpoint = $0 }
+            .overlay(alignment: .top) {
+                LabeledContent("Minimum population") {
+                    TextField("1,000,000", value: $minimumPopulation, format: .number)
+                        .multilineTextAlignment(.trailing)
+                }
+                .padding(8)
+                .background(Color.primary.colorInvert())
+                .clipShape(.rect(cornerRadius: 5))
+                .shadow(radius: 50)
+                .padding()
+            }
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    Button("Query") {
+                        isQuerying = true
+                    }
+                    .disabled(isQuerying || minimumPopulation == nil || viewpoint == nil)
+                    .task(id: isQuerying) {
+                        guard isQuerying, let minimumPopulation, let viewpoint else {
+                            return
+                        }
+                        defer { isQuerying = false }
+                        
+                        do {
+                            try await model.queryMapImageSublayers(
+                                minimumPopulation: minimumPopulation,
+                                geometry: viewpoint.targetGeometry
+                            )
+                        } catch {
+                            self.error = error
+                        }
+                    }
+                }
+            }
+            .task {
+                // Sets up the sublayers when the sample appears.
+                do {
+                    try await model.setUpMapImageSublayers()
+                } catch {
+                    self.error = error
+                }
+            }
+            .errorAlert(presentingError: $error)
+    }
+}
+
+/// The view model for this sample.
+@MainActor
+private final class Model {
```

Comment:
```suggestion
@MainActor
@Observable
private final class Model {
```
Should the model be observable?

Reply:
I don't believe so since there aren't any properties that need observing (e.i., there is no view state dependent on the model). The reason I went with a model despite that was to separate out the business logic and make the sample a little more readable.